rstream/client.py

```diff
@@ -206,9 +227,12 @@ async def _listener(self) -> None:
 
             for _, handler in self._handlers.get(frame.__class__, {}).items():
                 try:
-                    maybe_coro = handler(frame)
-                    if maybe_coro is not None:
-                        await maybe_coro
+                    if frame.__class__ == schema.Deliver:
```

Comment:
This special handling of Deliver frame feels kinda hacky, maybe move this to `Consumer._on_deliver`?